src/object.d

```diff
@@ -1101,7 +1101,7 @@ class TypeInfo_Struct : TypeInfo
                     this.initializer().length == s.initializer().length;
     }
 
-    override size_t getHash(in void* p) @safe pure nothrow const
+    override size_t getHash(in void* p) @trusted pure nothrow const
```

Comment:
The @safe was equally wrong. This PR does **not** address (or change) the interface to getHash(), the change was made to reflect the unsafe operation inside the function. The issue you brought up is with the interface to getHash(), not the interface to hashOf().

> your responsibility

It is the responsibility of people who find bugs to post them to bugzilla. The bug you found has nothing to do with this PR, which is to fix the interface to `hashOf()`. You should take the credit for finding the issue by posting it to bugzilla.

Meanwhile, this PR is **only** concerned with the interface to `hashOf`.